generators/server/templates/src/test/java/package/web/rest/PublicUserResourceIT.java.ejs

```diff
@@ -0,0 +1,277 @@
+<%_ } _%>
+import org.springframework.beans.factory.annotation.Autowired;
+<%_ if (!reactive) { _%>
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+<%_ } else { _%>
+import org.springframework.boot.test.autoconfigure.web.reactive.AutoConfigureWebTestClient;
+<%_ } _%>
+import org.springframework.boot.test.context.SpringBootTest;
+<%_ if (cacheManagerIsAvailable === true) { _%>
+import org.springframework.cache.CacheManager;
+<%_ } _%>
+<%_ if (cacheProvider === 'memcached' ) { _%>
+import org.springframework.cache.support.NoOpCacheManager;
+<%_ } _%>
+import org.springframework.http.MediaType;
+<%_ if (databaseType === 'couchbase') { _%>
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.test.context.TestSecurityContextHolder;
+<%_ } _%>
+import org.springframework.security.test.context.support.WithMockUser;
+<%_ if (reactive) { _%>
+import <%= packageName %>.service.dto.<%= asDto('User') %>;
+import org.springframework.test.web.reactive.server.WebTestClient;
+<%_ } _%>
+<%_ if (!reactive) { _%>
+import org.springframework.test.web.servlet.MockMvc;
+<%_ } _%>
+<%_ if (databaseType === 'sql' && !reactive) { _%>
+import org.springframework.transaction.annotation.Transactional;
+<%_ } _%>
+<%_ if (reactive && searchEngine === 'elasticsearch') { _%>
+import reactor.core.publisher.Mono;
+<%_ } _%>
+
+<%_ if (databaseType === 'sql' && !reactive) { _%>
+import javax.persistence.EntityManager;
+<%_ } _%>
+<%_ if (databaseType === 'cassandra' || databaseType === 'couchbase') { _%>
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+<%_ } _%>
+<%_ if (databaseType === 'cassandra') { _%>
+import java.util.UUID;
+<%_ } _%>
+
+<%_ if (reactive) { _%>
+import static org.assertj.core.api.Assertions.assertThat;
+<%_ } _%>
+<%_ if (reactive && searchEngine === 'elasticsearch') { _%>
+import static org.mockito.Mockito.*;
+<%_ } _%>
+<%_ if (!reactive) { _%>
+    <%_ if (databaseType === 'sql' || databaseType === 'mongodb' || databaseType === 'neo4j' || databaseType === 'couchbase') { _%>
+import static org.hamcrest.Matchers.hasItems;
+    <%_ } _%>
+import static org.hamcrest.Matchers.hasItem;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+<%_ } else if (testsNeedCsrf) { _%>
+import static org.springframework.security.test.web.reactive.server.SecurityMockServerConfigurers.csrf;
+<%_ } _%>
+
+/**
+ * Integration tests for the {@link UserResource} REST controller.
+ */
+<%_ if (!reactive) { _%>
+@AutoConfigureMockMvc
+<%_ } else { _%>
+@AutoConfigureWebTestClient
+<%_ } _%>
+@WithMockUser(authorities = AuthoritiesConstants.ADMIN)
+<%_ if (authenticationType === 'oauth2') { _%>
+@SpringBootTest(classes = {<%= mainClass %>.class, TestSecurityConfiguration.class})
+<%_ } else { _%>
+@SpringBootTest(classes = <%= mainClass %>.class)
```

Comment:
can be replaced by `@IntegrationTest`, and all other `@ExtendWith` can be removed

Reply:
should be resolved by https://github.com/jhipster/generator-jhipster/pull/13326